vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -555,14 +554,28 @@ public GridDataView<T> getGenericDataView() {
      * @return the created hierarchy column
      */
     public Column<T> addHierarchyColumn(ValueProvider<T, ?> valueProvider) {
-        Column<T> column = addColumn(TemplateRenderer
-                .<T> of("<vaadin-grid-tree-toggle "
-                        + "leaf='[[!item.children]]' expanded='{{expanded}}' level='[[level]]'>[[item.name]]"
-                        + "</vaadin-grid-tree-toggle>")
+
+        // The click listener needs to check if the event gets canceled (by
+        // vaadin-grid-tree-toggle) and only invoke the callback if it does.
+        // vaadin-grid-tree-toggle will cancel the event if the user clicks on
+        // a non-focusable element inside the toggle.
```

Comment:
Looks like we are not doing that here though. Compared to `HierarchyColumnRenderer`:
https://github.com/vaadin/flow-components/blob/158b50b10a3acd9a5b104d738c4015d284e0be39/vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/HierarchyColumnComponentRenderer.java#L41-L45

From that description it looks like it's not needed here, as we will always only have text node as content. Should we remove the comment then?

Reply:
Oops. I first had the same check here but also noticed it's not needed so removed the logic. Apparently forgot to remove the comment. Done.

Reply:
That's odd. I pushed the commit but it doesn't show in GH yet